### PR TITLE
Add support for prometheus servicemonitor

### PR DESCRIPTION
**Pull request by mobasherul**

### CODE REVIEW

**Comment by mobasherul** [comment]
FYI - I tested with non secure tmc it is working but not with secure tmc. ServiceMonitor might need more changes to support secure tmc.

**Comment by mathieucarbou** [comment]
> FYI - I tested with non secure tmc it is working but not with secure tmc. ServiceMonitor might need more changes to support secure tmc.

Its the same issue with ingress...

**Comment by mathieucarbou** [comment]
@thomas-2020 : do you want to review this PR since you opened the issue ?

**Review by mathieucarbou — APPROVED** [review]

**Review by mathieucarbou — CHANGES_REQUESTED** [review]
@mobasherul : could you please update the chart version:
`version: "1.2.1"` => `version: "1.3.0"` ?

**Review by mathieucarbou — CHANGES_REQUESTED** [review]
Could you also please run:

`docker run --rm --volume "$(pwd):/helm-docs" -u $(id -u) jnorwood/helm-docs:latest`

to update the `README.md` (for the table)

File: terracottabigmemorymax/helm/values.yaml [hunk 1]
```diff
@@ -76,6 +76,8 @@ terracotta:
   tmcOpts: ""
   # -- The license content for the Terracotta cluster. Optional.
   license: ""
+  # -- Can be used for enabling prometheus operator service monitor for scraping metrics.
+  serviceMonitorEnabled: false
```
Comment (thomas-2020) [inline terracottabigmemorymax/helm/values.yaml:80]: Is it duplicated? see `serviceMonitor` in line 153

**Comment by thomas-2020** [comment]
> Could you also please run:
> 
> `docker run --rm --volume "$(pwd):/helm-docs" -u $(id -u) jnorwood/helm-docs:latest`
> 
> to update the `README.md` (for the table)



> Could you also please run:
> 
> `docker run --rm --volume "$(pwd):/helm-docs" -u $(id -u) jnorwood/helm-docs:latest`
> 
> to update the `README.md` (for the table)

The action should be run automatically after commit in a "non" main branch. The last run was last Friday. When did you commit your branch?

**Review by thomas-2020 — CHANGES_REQUESTED** [review]
In `values.yaml`are multiple `servicesMonitors` to disable/enable.

File: terracottabigmemorymax/helm/values.yaml (hunk 1 above)
Comment (mobasherul) [reply terracottabigmemorymax/helm/values.yaml:80]: need to remove this

**Comment by mathieucarbou** [comment]
> after commit in a "non" main branch

That is the problem.

We don't create temporary remote branches to avoid creating a lot of "zombie" branches in the upstream repo, especially for a specific need in a specific point in time for that. This one is just a PR sent from a fork.

Ideally,  the Github workflow would need to be updated to either validate and check a PR (to make sure the generation was done), or trigger this generation each time a merge occurs in main and then create a PR with the changes. That's what I am doing on my other OSS projects to either generate snapshot documentation or release notes.

**Review by thomas-2020 — APPROVED** [review]
it looks good

**Review by mathieucarbou — APPROVED** [review]

**Pull request merged**